examples/go/tagging/handler/handler_test.go

```diff
@@ -0,0 +1,197 @@
+const passMark = "\u2713"
+const failMark = "\u2717"
+
+// TestLoadTomlCfg shows valid vcconfig.toml files can be loaded and processed.
+func TestLoadTomlCfg(t *testing.T) {
+	var tests = []struct {
+		testDesc  string
+		cfgPath   string
+		expectErr bool
+		want      *vcConfig
+	}{
+		{
+			"Test that toml file loads correctly",
+			"testdata/vcconfig.toml",
+			false,
+			&vcConfig{
+				struct {
+					Server   string
+					User     string
+					Password string
+					Insecure bool
+				}{
+					"veba.local.corp",
+					"admin@vsphere.local",
+					"password1234",
+					false,
+				},
+				struct {
+					URN    string
+					Action string
+				}{
+					"urn:vmomi:InventoryServiceTag:11f16f36-f5c4-4c29-b7d3-d9c7d12babe6:GLOBAL",
+					"attach",
+				},
+			},
+		},
+		{
+			"Test that toml file loads, even with more info than needed, and defaults are set",
+			"testdata/vcconfig2.toml",
+			false,
+			&vcConfig{
+				struct {
+					Server   string
+					User     string
+					Password string
+					Insecure bool
+				}{
+					"veba.local.corp",
+					"admin@vsphere.local",
+					"password1234",
+					true,
+				},
+				struct {
+					URN    string
+					Action string
+				}{
+					"urn:vmomi:InventoryServiceTag:11f16f36-f5c4-4c29-b7d3-d9c7d12babe6:GLOBAL",
+					"detach",
+				},
+			},
+		},
+		{
+			"Test that misconfigured toml file ends in error",
+			"testdata/vcconfigErr1.toml",
+			true,
+			nil,
+		},
+		{
+			"Test that vcconfig.toml missing essential information results in error.",
+			"testdata/vcconfigErr2.toml",
+			true,
+			nil,
+		},
+		{
+			"Test that missing toml file results in error",
+			"testdata/missing.toml",
+			true,
+			nil,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Logf("=========== %v ===========", tc.testDesc)
+		cfg, err := loadTomlCfg(tc.cfgPath)
+		if err != nil {
+			if tc.expectErr {
+				// An error is expected.
+				t.Logf("got an error, as expected: %v. %v", err, passMark)
+			} else {
+				t.Log(tc.testDesc, failMark, err)
+				t.Fail()
+			}
+		} else {
+			if *cfg == *tc.want {
+				t.Logf("got expected: %v. %v", tc.want, passMark)
+			} else {
+				t.Logf("expected: %v, got: %v. %v", tc.want, cfg, failMark)
+				t.Fail()
+			}
+		}
+
+	}
+}
+
+// TestParseEventMoRef ensures that managed object reference value and type are
+// obtained by the event json that meets Cloud Event specifications.
+func TestParseEventMoRef(t *testing.T) {
+	type vm struct {
+		vmType string
+		Value  string
+	}
+
+	var tests = []struct {
+		testDesc  string
+		jsonPath  string
+		expectErr bool
+		want      vm
+	}{
+		{
+			"Test that event is readable",
+			"testdata/event.json",
+			false,
+			vm{vmType: "VirtualMachine", Value: "vm-10000"},
+		},
+		{
+			"Event should be readable, even with minimal information",
+			"testdata/event2.json",
+			false,
+			vm{vmType: "VirtualMachine", Value: "vm-2"},
+		},
+		{
+			"Event should return error if VM type and value are null",
+			"testdata/eventErr1.json",
+			true,
+			vm{},
```

Comment:
In case of error these should be nil not empty struct as per your function signature. The test still succeeds because you don't check for the moref result in that case.